app-containers/lxd/lxd-5.20.ebuild

```diff
@@ -0,0 +1,188 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit bash-completion-r1 go-module linux-info optfeature systemd verify-sig
+
+DESCRIPTION="Modern, secure and powerful system container and virtual machine manager"
+HOMEPAGE="https://ubuntu.com/lxd https://github.com/canonical/lxd"
+SRC_URI="https://github.com/canonical/lxd/releases/download/${P}/${P}.tar.gz
+	verify-sig? ( https://github.com/canonical/lxd/releases/download/${P}/${P}.tar.gz.asc
+)"
+
+LICENSE="Apache-2.0 BSD LGPL-3 MIT"
+SLOT="0/stable"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="apparmor nls"
+
+DEPEND="acct-group/lxd
+	app-arch/xz-utils
+	>=app-containers/lxc-5.0.0:=[apparmor?,seccomp(+)]
+	dev-db/sqlite:3
+	>=dev-libs/dqlite-1.13.0:=
+	dev-libs/lzo
+	>=dev-libs/raft-0.17.1:=[lz4]
+	>=dev-util/xdelta-3.0[lzma(+)]
+	net-dns/dnsmasq[dhcp]
+	sys-libs/libcap
+	virtual/udev"
+RDEPEND="${DEPEND}
+	|| (
+			net-firewall/nftables[json]
+			(
+				net-firewall/ebtables
+				net-firewall/iptables
+			)
+	)
+	sys-apps/iproute2
+	sys-fs/fuse:3
+	>=sys-fs/lxcfs-5.0.0
+	sys-fs/squashfs-tools[lzma]
+	virtual/acl"
+BDEPEND="dev-lang/go
```

Comment:
This needs to be >=1.20 now

Reply:
We haven't had <1.20 in tree for months, wouldn't worry about it.

Reply:
It matters for upgrade ordering.

Reply:
I'm still of the opinion that this is not an issue, but whatever, the fix is simple and non-breaking: a151520795aa968794db41f60af6bf2821eef84f

@listout please do check the upstream announcement in future. They're usually very thorough in documenting bigger changes! 
(Also somewhat nice to see commits from incus the fork being incorporated into lxd, looks healthier than I thought)

Reply:
It is an issue, and it'd be great to not have to debate it every time.

If you have any system which happens to have Go 1.19 (1.20 was only released in Feb!) and lxd, if you have lxd scheduled before go, you will end up with emerge aborting. That's _particularly_ bad if you have something like a merge order like:
* python-exec
* lxd (breaks here)
* go (can't get to this point, and emerge is broken because portage didn't get re-emerged yet)
* python
* portage

Having the constraint costs nothing and takes no time at all to add, but can save people a lot of time by avoiding something getting bricked, and it's also just.. accurate.

Reply:
My opinion still is that if you're running ~unstable lxd and a system last updated a year ago, this is the least of your concerns out there. But yes the fix is cheap and other people are trying hard to make the possible upgrade as smooth as possible, so I should try a bit harder too.

Now offtopic. I said: 

> (Also somewhat nice to see commits from incus the fork being incorporated into lxd, looks healthier than I thought)

which didn't age too good: https://stgraber.org/2023/12/12/lxd-now-re-licensed-and-under-a-cla/

what a mess.